oqs-template/generate.py

```diff
@@ -193,25 +193,24 @@ def load_config(include_disabled_sigs=False):
 
     for kem in config['kems']:
         kem['hybrids'] = []
-        try:
-            for extra_nid_current in kem['extra_nids']['current']:
-                extra_hybrid = extra_nid_current
-                if extra_nid_current['hybrid_group'] == "x25519" or extra_nid_current['hybrid_group'] == "p256":
-                   extra_hybrid['bit_security']=128
-                if extra_nid_current['hybrid_group'] == "x448" or extra_nid_current['hybrid_group'] == "p384":
-                   extra_hybrid['bit_security']=192
-                if extra_nid_current['hybrid_group'] == "p521":
-                   extra_hybrid['bit_security']=256
-                kem['hybrids'].append(extra_hybrid)
-                if 'hybrid_group' in extra_nid_current and extra_nid_current['hybrid_group'] in ["x25519", "x448"]:
-                    extra_hyb_nid = extra_nid_current['nid']
-                    if 'nid_ecx_hybrid' in kem:
-                        print("Warning, duplicate nid_ecx_hybrid for",
-                              kem['name_group'], ":", extra_hyb_nid, "in generate.yml,",
-                              kem['nid_ecx_hybrid'], "in generate_extras.yml, using generate.yml entry.")
-                    kem['nid_ecx_hybrid'] = extra_hyb_nid
-        except KeyError as ke:
-            pass
+        if 'extra_nids' not in kem or 'current' not in kem['extra_nids']:
+            continue
+        hybrid_nids = set()
+        for extra_hybrid in kem['extra_nids']['current']:
+            if extra_hybrid['hybrid_group'] == "x25519" or extra_hybrid['hybrid_group'] == "p256":
+               extra_hybrid['bit_security'] = 128
+            if extra_hybrid['hybrid_group'] == "x448" or extra_hybrid['hybrid_group'] == "p384":
+               extra_hybrid['bit_security'] = 192
+            if extra_hybrid['hybrid_group'] == "p521":
+               extra_hybrid['bit_security'] = 256
+            kem['hybrids'].append(extra_hybrid)
+            if 'hybrid_group' in extra_hybrid:
+                extra_hybrid_nid = extra_hybrid['nid']
+                if extra_hybrid_nid in hybrid_nids:
+                    print("Warning, duplicate hybrid NID for",
+                          kem['name_group'], ":", extra_hybrid_nid, "in generate.yml.")
+                    exit(1)
```

Comment:
This `exit` warrants a bit more than a "Warning" message, no? I'd suggest to either remove the exit() or rephrase the Warning to "Fatal Error" (best with explanation what to change in the YML to get the generator to run again).

Reply:
Fair enough. This seems like it should be a fatal condition so I'll update the message to reflect. Please correct me if this is not the case.